### PR TITLE
Adds integration hooks to many SSI functions

### PR DESCRIPTION
**Pull request by Sesquipedalian**
It would be nice to be able to access many of the existing SSI functions via integration hooks, so this adds a bunch. (Not every SSI function gets one, of course; just the ones where it makes sense.)

In general, call_integration_hook() has been positioned so that it passes a given SSI function's output to the hook just before returning the output, so that the hooked code can make any final changes it needs to.

For the ssi_todaysCalendar() variants (ssi_todaysBirthdays, ssi_todaysHolidays, ssi_todaysEvents, and ssi_todaysCalendar), the same hook is used in each case, and call_integration_hook() simply passes on the $eventOptions array as a parameter so that the hooked code can handle different cases as needed.

For ssi_news(), nothing is passed to the hook, because the only relevant data is already in $context.

### CODE REVIEW

File: SSI.php [hunk 1]
```diff
@@ -998,6 +1013,9 @@ function ssi_queryMembers($query_where = null, $query_where_params = array(), $q
 
 	// Load the members.
 	loadMemberData($members);
+	
+	// If mods want to do somthing with this list of members, let them do that now.
+	call_integration_hook('integrate_ssi_queryMembers', array(&$members));
```
Comment (live627) [inline SSI.php]: this should be before loadMemberData

Comment (Sesquipedalian) [inline SSI.php]: Ah, yes. You are right.

File: SSI.php [hunk 2]
```diff
@@ -1312,6 +1336,9 @@ function ssi_recentPoll($topPollInstead = false, $output_method = 'echo')
 	}
 
 	$return['allowed_warning'] = $row['max_votes'] > 1 ? sprintf($txt['poll_options6'], min(count($sOptions), $row['max_votes'])) : '';
+	
+	// If mods want to do somthing with this list of polls, let them do that now.
+	call_integration_hook('integrate_ssi_recentPoll', array(&$return, $topPollInstead));
```
Comment (live627) [inline SSI.php:1341]: this should be before $return['allowed_warning']

Derp. nvm.

**Comment by live627** [comment]
Please sign off your commits.

File: SSI.php (hunk 2 above)
Comment (Sesquipedalian) [inline SSI.php:1341]: Really? I don't really get why...

Edit: Fair enough, I'll nvm. :)

**Comment by Sesquipedalian** [comment]
I don't much like the idea of publicly posting my raw email address for spam bots to pick up, but if the ... [at] ... [dot] ... convention isn't good enough to count as signing off on my code, alright then.

**Comment by live627** [comment]
Signing the PR isn't good enough, sorry...

**Comment by Sesquipedalian** [comment]
Fixed now.

**Comment by live627** [comment]
Ah thanks.

**Pull request merged**